src/cpp/enclave/CMakeLists.txt

```diff
@@ -33,10 +31,38 @@ if ("$ENV{DEBUG}" EQUAL 1)
   target_compile_definitions(enclave_trusted PUBLIC -DDEBUG)
 endif()
 
-add_compile_options(${COMMON_COMPILE_FLAGS} -fvisibility=hidden -fpie -fstack-protector -Wl,--no-undefined -nostdlib -nodefaultlibs -nostartfiles -Wl,-Bstatic -Wl,-Bsymbolic -Wl,--no-undefined -Wl,-pie -Wl,--export-dynamic -Wl,--defsym,__ImageBase=0)
+add_compile_options(${COMMON_COMPILE_FLAGS} -fvisibility=hidden -fpie -fstack-protector -Wno-unused-parameter -Wno-unused-variable -Wl,--no-undefined -nostdlib -nodefaultlibs -nostartfiles -Wl,-Bstatic -Wl,-Bsymbolic -Wl,--no-undefined -Wl,-pie -Wl,--export-dynamic -Wl,--defsym,__ImageBase=0)
 target_compile_definitions(enclave_trusted PUBLIC OE_API_VERSION=2)
 set_target_properties(enclave_trusted PROPERTIES POSITION_INDEPENDENT_CODE OFF)
 
+########## Fetch and make opaque-utils library available ##########
+include(FetchContent)
+FetchContent_Declare(
+  mc2_utils_e
+  GIT_REPOSITORY https://github.com/mc2-project/mc2-utils.git
+)
+set(FETCHCONTENT_QUIET OFF)
+set(HOST OFF CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(mc2_utils_e)
+add_compile_definitions(SPDLOG_NO_THREAD_ID FMT_USE_INT128=0)
+
+include_directories(${spdlog_SOURCE_DIR}/include)
+##################################################################
+
+########## Fetch and make mc2_serialization library available ##########
+include(FetchContent)
+FetchContent_Declare(
+  mc2_serialization
+  GIT_REPOSITORY https://github.com/mc2-project/mc2-serialization.git
+)
+set(FETCHCONTENT_QUIET OFF)
+FetchContent_MakeAvailable(mc2_serialization)
+
+message(ERROR ${CMAKE_CURRENT_BINARY_DIR})
+#TODO(octavian): remove this once the serialization library allows direct includes
+include_directories(${CMAKE_CURRENT_BINARY_DIR}/../_deps/mc2_serialization-src/src)
```

Comment:
@ryanleh for some reason the serialization library does not allow me to directly include using `#include "serialization.h`